pkg/controller/utils/component.go

```diff
@@ -328,6 +330,14 @@ func ensureOSSchedulingRestrictions(obj client.Object, osType rmeta.OSType) {
 		for i := range nodeSets {
 			podSpecs = append(podSpecs, &nodeSets[i].PodTemplate.Spec)
 		}
+	case *monitoringv1.Alertmanager:
+		// Prometheus operator types don't have a template spec which is of v1.PodSpec type.
+		podSpec := &obj.(*monitoringv1.Alertmanager).Spec
+		podSpec.NodeSelector = map[string]string{"kubernetes.io/os": string(osType)}
+	case *monitoringv1.Prometheus:
+		// Prometheus operator types don't have a template spec which is of v1.PodSpec type.
+		podSpec := &obj.(*monitoringv1.Prometheus).Spec
+		podSpec.NodeSelector = map[string]string{"kubernetes.io/os": string(osType)}
```

Comment:
This line seems redundant. Won't this be done in the for-loop below as well?

Reply:
`Alertmanager` and `Prometheus` types don't have a template podsepc like others that can be added to `podSpecs`. I could have returned from here.

Reply:
Ah, now I see what you mean.